File: produtos/src/main/java/br/com/bota/entity/listener/ProdutoListener.java
package br.com.bota.entity.listener;

import br.com.bota.entity.Produto;

import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;
import java.time.LocalDateTime;

public class ProdutoListener {
    @PrePersist
    public void onSave(Produto product) {
        product.setCriadoEm(LocalDateTime.now());
    }

    @PreUpdate
    public void onUpdate(Produto product) {
        product.setModificadoEm(LocalDateTime.now());
    }
}

File: loja-kotlin/kprodutos/src/main/kotlin/br/com/bota/entity/listener/ProdutoListener.kt
package br.com.bota.entity.listener

import br.com.bota.entity.Produto
import java.time.LocalDateTime
import javax.persistence.PrePersist
import javax.persistence.PreUpdate


open class ProdutoListener {
    @PrePersist
    open fun onSave(produto: Produto) {
        produto.criadoEm = LocalDateTime.now()
    }

    @PreUpdate
    open fun onUpdate(produto: Produto) {
        produto.modificadoEm = LocalDateTime.now()
    }
}